addon/components/polaris-resource-list/bulk-actions.js

```diff
@@ -0,0 +1,535 @@
+      } = this.getProperties(
+        'promotedActions',
+        'numberOfPromotedActionsToRender',
+        'measuring'
+      );
+
+      if (
+        !promotedActions ||
+        (promotedActions && numberOfPromotedActionsToRender === 0 && !measuring)
+      ) {
+        return 'Actions';
+      }
+
+      return 'More actions';
+    }
+  ),
+
+  shouldRenderActionsPopover: computed(
+    'actionSections',
+    'rolledInPromotedActions.length',
+    'measuring',
+    function() {
+      let {
+        actionSections,
+        rolledInPromotedActions,
+        measuring,
+      } = this.getProperties(
+        'actionSections',
+        'rolledInPromotedActions',
+        'measuring'
+      );
+
+      return Boolean(
+        actionSections || rolledInPromotedActions.length > 0 || measuring
+      );
+    }
+  ),
+
+  actionSections: computed('actionsCollection.[]', function() {
+    let actionsCollection = this.get('actionsCollection');
+
+    if (!actionsCollection || actionsCollection.length === 0) {
+      return null;
+    }
+
+    if (this.instanceOfBulkActionListSectionArray(actionsCollection)) {
+      return actionsCollection;
+    }
+
+    if (this.instanceOfBulkActionArray(actionsCollection)) {
+      return [{ items: actionsCollection }];
+    }
+  }),
+
+  combinedActions: computed(
+    'actionSections.[]',
+    'rolledInPromotedActions',
+    function() {
+      let combinedActions = [];
+      let { actionSections, rolledInPromotedActions } = this.getProperties(
+        'actionSections',
+        'rolledInPromotedActions'
+      );
+
+      let rolledInPromotedActionsHasLength = rolledInPromotedActions.length > 0;
+
+      if (actionSections && rolledInPromotedActionsHasLength) {
+        combinedActions = [
+          { items: rolledInPromotedActions },
+          ...actionSections,
+        ];
+      } else if (actionSections) {
+        combinedActions = actionSections;
+      } else if (rolledInPromotedActionsHasLength) {
+        combinedActions = [{ items: rolledInPromotedActions }];
+      }
+
+      return combinedActions;
+    }
+  ),
+
+  /**
+   * Callback when selectable state of list is changed
+   *
+   * @type {Function}
+   * @default noop
+   * @property onSelectModeToggle
+   */
+  onSelectModeToggle(/* selectMode **/) {},
+
+  /**
+   * Callback when the select all checkbox is clicked
+   *
+   * @type {Function}
+   * @default noop
+   * @property onToggleAll
+   */
+  onToggleAll() {},
+
+  instanceOfBulkActionListSectionArray(actionsCollection) {
+    let validList = actionsCollection.filter((action) => {
+      return action.items;
+    });
+
+    return actionsCollection.length === validList.length;
+  },
+
+  instanceOfBulkActionArray(actionsCollection) {
+    let validList = actionsCollection.filter((action) => {
+      return !action.items;
+    });
+
+    return actionsCollection.length === validList.length;
+  },
+
+  setContainerNode() {
+    this.set('containerNode', this.element);
+  },
```

Comment:
I don't see any real point to doing this - if you want to set `containerNode` to `this.element`, maybe use `alias`? Or we could just use `this.element` everywhere the React code uses `this.containerNode`, aliasing it seems a bit pointless 🤷‍♂️

Reply:
Ah I hadn't thought of that, yeah lets just use `this.element`